### PR TITLE
numpy doc enhancement

### PR DESCRIPTION
**Pull request by reminisce**
1. Enhanced docs for numpy ops.
2. Added operators `np.nonzero` and `np.argmin`.
3. Fixed https://github.com/apache/incubator-mxnet/issues/16642

### CODE REVIEW

File: python/mxnet/numpy/linalg.py [hunk 1]
```diff
@@ -54,10 +54,42 @@ def norm(x, ord=None, axis=None, keepdims=False):
...
+    >>> b = a.reshape((3, 3))
+    >>> b
+    array([[-4., -3., -2.],
+           [-1.,  0.,  1.],
+           [ 2.,  3.,  4.]])
+    >>>
+    >>>
+    >>>
+    >>>
+    >>>
```
Comment (haojin2) [inline python/mxnet/numpy/linalg.py]: Why extra blank lines here?

File: python/mxnet/numpy/multiarray.py [hunk 2]
```diff
@@ -2358,6 +2430,18 @@ def subtract(x1, x2, out=None, **kwargs):
     -------
     subtract : ndarray or scalar
         The difference of x1 and x2, element-wise. This is a scalar if both x1 and x2 are scalars.
+
+    Examples
+    --------
+    >>> np.subtract(1.0, 4.0)
+    -3.0
+
```
Comment (haojin2) [inline python/mxnet/numpy/multiarray.py]: No extra blank line here?

**Comment by haojin2** [comment]
Please fix the sanity build first

File: tests/python/unittest/test_numpy_interoperability.py [hunk 3]
```diff
@@ -349,6 +349,22 @@ def _add_workload_argmax():
...
+def _add_workload_argmin():
+    OpArgMngr.add_workload('argmin', np.random.uniform(size=(4, 5, 6, 7, 8)), 0)
+    OpArgMngr.add_workload('argmin', np.random.uniform(size=(4, 5, 6, 7, 8)), 1)
+    OpArgMngr.add_workload('argmin', np.random.uniform(size=(4, 5, 6, 7, 8)), 2)
+    OpArgMngr.add_workload('argmin', np.random.uniform(size=(4, 5, 6, 7, 8)), 3)
+    OpArgMngr.add_workload('argmin', np.random.uniform(size=(4, 5, 6, 7, 8)), 4)
+    # OpArgMngr.add_workload('argmin', np.array([0, 1, 2, 3, np.nan]))
+    # OpArgMngr.add_workload('argmin', np.array([0, 1, 2, np.nan, 3]))
+    # OpArgMngr.add_workload('argmin', np.array([np.nan, 0, 1, 2, 3]))
+    # OpArgMngr.add_workload('argmin', np.array([np.nan, 0, np.nan, 2, 3]))
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_interoperability.py]: Are those test cases failing at this moment?

File: python/mxnet/numpy/linalg.py (hunk 1 above)
Comment (reminisce) [reply python/mxnet/numpy/linalg.py]: Copied from numpy doc to command line and ran from there, but seems redundant here. I just deleted them.

File: python/mxnet/numpy/multiarray.py (hunk 2 above)
Comment (reminisce) [reply python/mxnet/numpy/multiarray.py]: Deleted.

File: tests/python/unittest/test_numpy_interoperability.py (hunk 3 above)
Comment (reminisce) [reply tests/python/unittest/test_numpy_interoperability.py]: Currently, mxnet does not have a consistent behavior with NumPy in dealing with `np.nan`. Will re-enable them after this problem is resolved.

**Pull request merged**